### ISSUE TITLE
New Rule: implement `markdownlint`'s `MD023`

### ISSUE THREAD

**Issue by tooth-is-silver** [eslint-markdown/eslint-markdown#742]
### Rule details

Add a rule equivalent to markdownlint's MD023.

The rule should report headings that are preceded by spaces instead of starting at the beginning of the line. Indentation introduced by blockquotes should remain valid.

Provide an autofix when the indentation can be removed safely.

### What type of rule is this?

Warns about a potential problem

### Example code

```markdown
<!-- Incorrect -->

Some text

  # Indented heading

<!-- Correct -->

Some text

# Heading

> # Heading in a blockquote
```

### Prior Art

- markdownlint: [`MD023` - Headings must start at the beginning of the line](https://github.com/DavidAnson/markdownlint/blob/main/doc/md023.md)

**Comment by sohxxny** [comment]
Hi! I'd like to work on this issue. Could I take it?

**Comment by tooth-is-silver** [comment]
Thanks for your interest! I've assigned this issue to @sohxxny
